src/JointTrajectoryControllerBase.cpp

```diff
@@ -349,6 +374,25 @@ void JointTrajectoryControllerBase::goalCallback(GoalHandle goalHandle)
   newContext->mTrajectory = trajectory;
   newContext->mGoalHandle = goalHandle;
 
+  // Evaluate the trajectory at the current time.
+  auto offsetDesiredState = mControlledSpace->createState();
+  Eigen::VectorXd offsetDesiredPosition(mControlledSpace->getDimension());
+  trajectory->evaluate(0, offsetDesiredState);
+  mCompoundSpace->logMap(offsetDesiredState, offsetDesiredPosition);
+
+  Eigen::VectorXd offsetActualPosition(mControlledSpace->getDimension());
+  offsetActualPosition = mControlledSkeleton->getPositions();
+
+  Eigen::VectorXd offset(mControlledSpace->getDimension());
+  offset = offsetDesiredPosition - offsetActualPosition;
+  
+  offset = (offset / (2*M_PI));
+  for (int i = 0; i < offset.size(); ++i)
+  {
+    offset(i) = round(offset(i))*2*M_PI;
+  }
+  mOffset = offset;
```

Comment:
`mOffset` is how much the trajectory's initial point is away from the current skeleton position in Rn space.  Can we give it a more descriptive name like, `mCurrentTrajectoryOffset`?